0017-negotiation-and-execution-protocol.adoc

```diff
@@ -0,0 +1,140 @@
+
+1. Solution proposal for splitting trade negotiation and execution in different protocols.
+2. Solution proposal for the "How to update an offer"/"Offers should not get stale" problem.
+3. Defining the base for a first negotiation protocol.
+4. Defining the changes needed to the existing execution protocol RFC003.
+
+== Research ==
+
+=== Definitions
+
+=== Trade ===
+
+In the COMIT context a trade is always between two digital assets.
+
+We identify two main phases when it comes to a trade:
+
+1. Negotiation: The phase where the maker comes up with one or multiple offers.
+The outcome of the negotiation phase should be the parameters needed for execution.
+2. Execution: The phase where the trade is executed, hence actions are executed on the ledger.
+
+=== Offer ===
+
+We define an "offer" as part of the negotiation phase of a trade.
+There can be multiple offers during the negotiation phase.
+Offers are created by the maker.
+
+=== Negotiation roles ===
+
+* Maker: Creates offers
+* Taker: Takes offers
+
+Note that these roles are orthogonal to the cryptographic roles (Alice and Bob) in the execution protocol.
+Both the maker and the taker can be in the role of Alice (who generates the secret) or Bob.
+During the negotiation phase both parties have to reach agreement on who will be in the role of Alice and who will be in the role of Bob.
+
+=== Assumptions
+
+==== Protocols
+For the sake of simplicity we assume that there is only one protocol (RFC003) for trade execution at the moment.
+In the future there will be more execution protocols.
+
+As we build upon libp2p for messaging all protocols should adhere to the libp2p specification. The proposed first negotiation protocol will be a new libp2p protocol.
+
+==== Offers not getting stale
+
+During the negotiation phase the offer can be changed multiple times.
+Once the negotiation phase comes to a successful end (the parties agree on the parameters for the execution) the execution phase is triggered (immediately).
+Note that there is still an implicit accept of the
+
+=== Recommendation for protocol changes
+
+The general idea is to first define what parameters are needed for execution phase and then design a negotiation protocol that outputs these parameters for the execution.
+
+As RFC003 is currently the only implementation of an execution protocol we focus on this protocol.
+Once more execution protocols are defined we will have to define more negotiation protocols according to the parameters required by the execution protocol.
+This means, that some negotiation protocols could be re-used for multiple execution protocols if the execution protocols have the same input parameters.
+
+==== Changes to the existing Execution protocol RFC002/RFC003
+
+===== Input parameters
+
+For the execution phase we could identify the following input parameters:
+1. PeerID
+2. Ledgers (alpha and beta ledger)
+3. Assets (alpha and beta asset)
+4. Amounts
+5. Protocol
+6. Role
+7. Offer ID
+
```

Comment:
This list does not render correctly.  Ordered list can be achieved using just the dot.